### ISSUE TITLE
Cancancan: namespaces or not

### ISSUE THREAD

**Issue by davidwessman** [fsek/web#355]
The reason I started using admin-controllers was that some controllers ended up as a mess of `if`statements and the same for the views. When we started splitting it into two controllers I thought the code got a lot more clean and easier to read (however sometimes the amount of files doubled).

### Rent
in the `normal controller` we allow users to create their own rents, edit the rents they own but also look at rents of others.

in the `admin controller` the administrators are allowed to see all rents, also create rents for people and so on.

If a user creates a `rent`, they get `can?(:edit, rent)`. If we would only use standard Cancancan in the admin controller they would be allowed to edit the rent through the admin controller as well.

## Alternatives
1. Use a `before_action :authorize` where `authorize(:manage, Model)` is called, since `manage` on a `Model` is not given to users through the `ability`model.
2. Get Cancancan to understand the admin namespace, I guess we would have to add an `AdminAbility`.

I realize that I have gotten very stuck into the first pattern and a lot of the code is written like this.

Reference:
https://github.com/CanCanCommunity/cancancan/wiki/Admin-Namespace

This says that you can namespace abilities, not sure if they only mean by aliases:
https://github.com/CanCanCommunity/cancancan/wiki/CanCan-2.0-Aliases#notes

**Comment by jforberg** [comment]
This is a real problem and using namespaces is a common solution. It's not my preferred solution as it tends to fracture the codebase (which is already super-fractured with the Rails standard file layout).

What I have done in the past is to simply filter away records which a user is not allowed to see. In the simplest case we can just use User#can? as a filter but sometimes that is too slow and a more complicated query has to be written.

The only case where this problem exists is in "viewing" actions. Modifying actions are usually done only on a single record so the problem disappears.

Here is an example of how I have done in the past:

```
def index
  @people = filter_records(Person.some_scope)
end

private
def filter_records arel
  arel.to_a.select{ |r| current_user.can? current_action, r }
end
```

Or for a more complicated use case:

```
def index
  @people = filter_records(Person.some_scope)
end

private
def filter_records arel
  # We need to know the person "department" to decide access, for whatever reason
  out = arel.includes(:department).to_a.select{ |r| current_user.department == r.department }
  # If no records remain, we call it an unauthorized access.
  out.any? else raise CanCan::AccessDenied('You shall not pass')
end
```

This has two advantages:
* Each action can use the same filter to ensure that no disallowed records are shown.
* We can hack the filter to do a more clever query if it turns out we need better performance. But we can also just leave it as is and let cancan do the heavy lifting.


If we are dead set on splitting our controllers, I'm sure cancan can be hacked (hehe) to support these namespaces. I can give it a shot if you are interested.

**Comment by davidwessman** [comment]
Closed by #358
